app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -706,34 +705,25 @@ class HomeFragment : Fragment() {
         }
     }
 
-    private fun showDeleteCollectionPrompt(
-        tabCollection: TabCollection,
-        title: String?,
-        message: String,
-        wasSwiped: Boolean,
-        handleSwipedItemDeletionCancel: () -> Unit
-    ) {
-        val context = context ?: return
-        AlertDialog.Builder(context).apply {
-            setTitle(title)
-            setMessage(message)
-            setNegativeButton(R.string.tab_collection_dialog_negative) { dialog: DialogInterface, _ ->
-                if (wasSwiped) {
-                    handleSwipedItemDeletionCancel()
-                }
-                dialog.cancel()
-            }
-            setPositiveButton(R.string.tab_collection_dialog_positive) { dialog: DialogInterface, _ ->
-                // Use fragment's lifecycle; the view may be gone by the time dialog is interacted with.
-                lifecycleScope.launch(IO) {
-                    context.components.core.tabCollectionStorage.removeCollection(tabCollection)
-                    context.components.analytics.metrics.track(Event.CollectionRemoved)
-                }.invokeOnCompletion {
-                    dialog.dismiss()
-                }
-            }
-            create()
-        }.show()
+    @VisibleForTesting
+    internal fun removeCollectionWithUndo(tabCollection: TabCollection) {
+        val snackbarMessage = getString(R.string.snackbar_collection_deleted)
+
+        lifecycleScope.allowUndo(
```

Comment:
I've never used `lifecycleScope.allowUndo` before. Will this cause any issues with blocking the main thread? should it be on IO instead?

Reply:
No, this won't block the main thread.  The only part that could block the main thread is end of `removeCollectionWithUndo` and it's launched in IO thread.  Thanks,